sshd-sftp/src/main/java/org/apache/sshd/server/subsystem/sftp/SftpEventListener.java

```diff
@@ -393,6 +393,30 @@ default void removed(ServerSession session, Path path, Throwable thrown) throws
         // ignored
     }
 
+    /**
+     * Called <U>prior</U> to removing a directory
+     *
+     * @param session The {@link ServerSession} through which the request was handled
+     * @param path    The {@link Path} about to be removed
+     * @throws IOException If failed to handle the call
+     * @see #removed(ServerSession, Path, Throwable)
+     */
+    default void removingDirectory(ServerSession session, Path path) throws IOException {
```

Comment:
In order to really make the distinction, the original `removing/remove` should also be renamed to `removingFile/removedFile`

Reply:
Ack